Assignment5/GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace GenericsHomework
+{
+    public class Node<T> : ICollection<T>
+    {
+        private T? _Data;
+
+        public T Data
+        {
+            get
+            {
+                return _Data!;
+            }
+            private set
+            {
+                _Data = value ??
+                    throw new ArgumentNullException(nameof(value));
+            }
+        }
+
+        private Node<T>? _Next;
+
+        public Node<T> Next
+        {
+            get
+            {
+                return _Next!;
+            }
+            private set
+            {
+                value._Next = this;
+                _Next = value ??
+                    throw new ArgumentNullException(nameof(value));
+            }
+        }
+
+        public int Count
+        {
+            get
+            {
+                return GetArrayLength();
+            }
+        }
+
+        public int GetArrayLength()
+        {
+            Node<T> cur = Next;
+            int count = 1;
+
+            while (cur != this)
+            {
+                cur = cur.Next;
+                count++;
+            }
+            return count;
+        }
+
+        public bool IsReadOnly
+        {
+            get;
+        }
+
+        public Node(T value)
+        {
+            Data = value;
+            Next = this;
+        }
+
+        public override string ToString()
+        {
+            if (Data is null) 
+                throw new ArgumentNullException(nameof(Data));
```

Comment:
```suggestion
            if (Data is null) 
                throw new ArgumentNullException(nameof(Data));
```
Consider returning an empty string if `Data` is null.